cmd/spire-agent/cli/run/run_test.go

```diff
@@ -861,6 +863,32 @@ func TestNewAgentConfig(t *testing.T) {
 				require.Nil(t, c)
 			},
 		},
+		{
+			msg: "warn_on_long_trust_domain",
+			input: func(c *Config) {
+				c.Agent.TrustDomain = strings.Repeat("a", 256)
+			},
+			logOptions: func(t *testing.T) []log.Option {
+				return []log.Option{
+					func(logger *log.Logger) error {
+						logger.SetOutput(ioutil.Discard)
+						hook := test.NewLocal(logger.Logger)
+						t.Cleanup(func() {
+							require.Len(t, hook.AllEntries(), 1)
+							entry := hook.LastEntry()
```

Comment:
it may be easier it you use something like:
```
         expectLogs := {
					Level:   logrus.WarnLevel,
					Message: "Configured trust domain should be less than 255 characters to be SPIFFE compliant",
					// All fields on actual entry.
					Data: logrus.Fields{},
	}
	spiretest.AssertLogs(t, test.logHook.AllEntries(), expectedLogs)
```

it will validates you have only 1 log and you can validate that the log entry is the one you expect with lvl, message and fields if you set one.